### DISCUSSION TITLE
going to wrong route

### DISCUSSION THREAD

**Discussion by thenaazimkhan** [vercel/next.js#46496] [Help]
Hi All,

I am facing an issue with my dynamic routing. Anyone can tell me what I am doing wrong.
in my pages folder, I have a discover folder.

Route. 1 -> /discover
Index page renders from the discover root.
screenshot
<img width="200" alt="Screenshot 2023-02-27 at 5 58 41 PM" src="https://user-images.githubusercontent.com/95960950/221583538-9d78cc29-c37d-41d9-b462-ff4693e4d3e8.png">

Route. 2 -> /discover/page/[page] => /discover/page/1,2,3...
The dynamic pagination page renders from folder /page/[page] in the discover root folder.
screenshot

<img width="208" alt="Screenshot 2023-02-27 at 6 28 42 PM" src="https://user-images.githubusercontent.com/95960950/221590977-a4dac56a-4f54-4657-8131-72bf5192b38f.png">

Route. 3 -> /discover/[category] => /discover/cat
The dynamic category index page renders from the [category] folder in the discover root folder.
screenshot

<img width="215" alt="Screenshot 2023-02-27 at 6 37 53 PM" src="https://user-images.githubusercontent.com/95960950/221592834-c61503b0-b597-4003-a51d-329ce6e22f5e.png">

Route. 4 -> /discover/[category]/[page] => /discover/cat/1, 2, 3....
The dynamic pagination [page] renders from the [category]/[page] folder in the discover root folder.
screenshot

<img width="213" alt="Screenshot 2023-02-27 at 6 42 28 PM" src="https://user-images.githubusercontent.com/95960950/221594015-db52f7fc-242e-4fcf-bf0e-8d92e2a73c1e.png">

All working fine but 
the problem is when I visit # Route 2 which is my dynamic route for the discover page. for example
/discover/page/1
/discover/page/2
till here also working fine and the page renders from /page/[page].
but when I visit 3, 4, and so on, my routing moved from the /page/[page] folder to /discover/[category]/[page] folder.

I do not understand this behavior. 
that should be renders from /page/[page] for all pagination or renders from /discover/[category]/[page]
why its render only 1 and 2 pages  from /page/[page] here and the rest from /discover/[category]/[page] here

anyone, please help me here, or suggest to me what I am doing wrong.

Nextjs version: 12.3.0
Nodejs version: 16.19.0

**Comment by icyJoseph** [comment]
Can you try to use the `as` prop on `next/link`? https://nextjs.org/docs/tag/v9.5.2/api-reference/next/link#dynamic-routes (from old docs)

**Reply by thenaazimkhan** [reply]
thanks, joseph for your reply. I found the issue, I know what I am doing wrong.

**Reply by icyJoseph** [reply]
Was it related to href/as when resolving routes?

Could you briefly describe what was the issue?

**Reply by thenaazimkhan** [reply]
it's due to the pagination per page size issue. I have 44 entries and 25 entries per page so only 2 pages were created.
when I visit /discover/page/1 and /discover/page/2 this will go to the correct file route (/discover/page/[page]) see screenshot at Route. 2 above.
now when I try to visit /discover/page/3 and /discover/page/4 and so on...
there are no more entries because 44 entries are covered in two pages. for this my discover/page/[page] path is not matched, and that is why is going to the Route. 4 -> /discover/[category]/[page].

**Reply by icyJoseph** [reply]
Ah! a business logic error, good catch! those can be tricky to realise. Good job.

**Reply by thenaazimkhan** [reply]
Yes, I didn't think so, :laughing: :laughing: I be like if it is working for pages 1 and 2 then have to work for 3,4... so on pages.
